tests/site-nav.js

```diff
@@ -75,33 +76,74 @@ describe('SiteNav', function() {
       $('body').width(1000);
     });
 
+    describe('initMobileMenu()', function() {
+      it('should append the mobile menu', function() {
+        expect(this.siteNav.$menu.find('.js-mobile-nav').length).to.equal(1);
+        expect(this.siteNav.isMobile).to.be.true;
+      });
+    });
+
     describe('assignAria()', function() {
       it('should assign aria attributes to the list and toggle', function() {
         expect(this.siteNav.$toggle.length).to.be.ok;
         expect(this.siteNav.$toggle.attr('aria-haspopup')).to.equal('true');
-        expect(this.siteNav.$list.attr('aria-hidden')).to.equal('true');
+        expect(this.siteNav.$menu.attr('aria-hidden')).to.equal('true');
       });
     });
 
     describe('toggle()', function() {
       function isOpen(siteNav) {
         return siteNav.isOpen &&
           siteNav.$body.hasClass('is-open') &&
-          siteNav.$list.attr('aria-hidden') === 'false' &&
+          siteNav.$menu.attr('aria-hidden') === 'false' &&
           siteNav.$toggle.hasClass('active');
       }
       function isClosed(siteNav) {
         return !siteNav.isOpen &&
           !siteNav.$body.hasClass('is-open') &&
-          siteNav.$list.attr('aria-hidden') !== 'false'  &&
+          siteNav.$menu.attr('aria-hidden') !== 'false'  &&
           !siteNav.$toggle.hasClass('active');
       }
+
       it('should show and hide the menu', function() {
-        this.siteNav.toggle();
+        this.siteNav.toggleMenu();
         expect(isOpen(this.siteNav)).to.be.true;
-        this.siteNav.toggle();
+        this.siteNav.toggleMenu();
         expect(isClosed(this.siteNav)).to.be.true;
       });
     });
+
+    describe('mobile panels', function() {
+      it('should show a panel when clicking a target', function() {
+        this.siteNav.$body.find('.js-panel-trigger[aria-controls="nav-advanced"]').click();
+        expect(this.siteNav.$body.find('#nav-advanced').attr('aria-hidden')).to.equal('false');
+      });
+
+      it('should hide a panel when clicking the back button', function() {
+        this.siteNav.$body.find('.js-panel-close[aria-controls="nav-advanced"]').click();
+        expect(this.siteNav.$body.find('#nav-advanced').attr('aria-hidden')).to.equal('true');
+      });
+    });
+  });
+
+  describe('switchMenu()', function() {
+    before(function() {
+      this.siteNav = new SiteNav('.js-site-nav');
+      $('body').width(400);
+    });
+
+    afterEach(function() {
+      $('body').width(1000);
+    });
```

Comment:
You have an imbalanced before/after, beforeEach/afterEach which on fist glance looks like a bug. I would split this up into two sub-`describe`s, one testing going from small to big, the other from big to small.

```javascript
describe('switchMenu()', function () {
  describe('when resizing small to large screen', function () {
    before(function() {
      // set mobile screen size
      // init .js-site-nav
      // assert mobile menu exists for sanity
      // set screen size large
    });
    after(function () {
      // restore to original size
    });
    it('has mega menu', function() {
      // assert mobile menu is removed
      // assert mega menu is in correct state
    });
  });

  describe('when resizing large to small screen', function () {
    // Similar to above
  });
});